alembic/versions/a5c06aa9ef30_add_runmodel.py

```diff
@@ -0,0 +1,621 @@
+            else:
+                run_models_successful += 1
+                for test, build in zip(test_group, matching_builds):
+                    if len(build.runs) != 1:
+                        PackitException(
+                            f"Build {build} does not have exactly one run:\n"
+                            f"{build.runs}"
+                        )
+                    test_runs_attached += 1
+                    build.runs[-1].test_run = test
+                    session.add(build.runs[-1])
+
+    srpm_builds_removed_for_no_job_trigger = 0
+    for srpm_build in session.query(SRPMBuildModel).all():
+        if not srpm_build.job_trigger:
+            srpm_builds_removed_for_no_job_trigger += 1
+            session.delete(srpm_build)
+
+    # Create RunModel for SRPMBuildModels without any build.
+    for job_trigger_model in session.query(JobTriggerModel).all():
+        if not job_trigger_model.copr_builds and not job_trigger_model.koji_builds:
+            for srpm_build in job_trigger_model.srpm_builds or []:
+                all_run_models += 1
+                srpms_without_build += 1
+                run_model = RunModel()
+                run_model.job_trigger = srpm_build.job_trigger
+                run_model.datetime = srpm_build.build_submitted_time
+                run_model.srpm_build = srpm_build
+                session.add(run_model)
+
+    print("================================")
+    print(f"SRPM models without any build: {srpms_without_build}")
+    print(
+        f"SRPM models removed because of no connection to any job trigger: "
+        f"{srpm_builds_removed_for_no_job_trigger}"
+    )
+    print("================================")
+    print(f"All Copr builds: {all_copr_builds}")
+    print(
+        f"Copr builds deleted for no SRPM for trigger: {deleted_copr_builds_for_no_srpm}"
+    )
+    print(f"Copr builds deleted for no SRPM set: {copr_builds_without_srpm}")
+    print(
+        f"Fixed SRPM matching to trigger model from Copr build: "
+        f"{fixed_srpm_matching_from_copr_build}"
+    )
+    print("================================")
+    print(f"All Koji builds: {all_koji_builds}")
+    print(
+        f"Koji builds deleted for no SRPM for trigger: {deleted_koji_builds_for_no_srpm}"
+    )
+    print(f"Koji builds deleted for no SRPM set: {koji_builds_without_srpm}")
+    print(
+        f"Fixed SRPM matching to trigger model from Koji build: "
+        f"{fixed_srpm_matching_from_koji_build}"
+    )
+    print("================================")
+    print(f"All Test runs: {all_test_runs}")
+    print(f"Attached correctly to build: {test_runs_attached}")
+    print(f"All Run models: {all_run_models}")
+    print(
+        f"Run models with different number of tests and builds: {number_of_builds_and_tests_differ}"
+    )
+    print(f"Run models with test run correctly set: {run_models_successful}")
+    print("================================")
+
+    # Check:
```

Comment:
I know that you run these on the prod data which is giving us a great confidence that the migration is gonna be smooth, but! since exceptions are being raised below, what should we do if one of the exceptions get fired?

Reply:
The transaction is not committed in that state so we will not change anything and can revert deployment and fix the migration script.